stdlib/public/core/StringBridge.swift

```diff
@@ -93,22 +51,52 @@ extension String {
   @inline(never) @_semantics("stdlib_binary_only") // Hide the CF dependency
   public // SPI(Foundation)
   init(_cocoaString: AnyObject) {
+    // If the NSString is actually a Swift String in disguise, 
+    // we can just copy out the internal repr.
     if let wrapped = _cocoaString as? _NSContiguousString {
       self._core = wrapped._core
       return
     }
 
     // "copy" it into a value to be sure nobody will modify behind
     // our backs.  In practice, when value is already immutable, this
-    // just does a retain.
+    // just does a retain. Even though we're unconditionally copying the bits
+    // into a new allocation, this is necessary to avoid races if others 
+    // are trying to mutate it.
     let cfImmutableValue
       = _stdlib_binary_CFStringCreateCopy(_cocoaString) as AnyObject
 
     let length = _swift_stdlib_CFStringGetLength(cfImmutableValue)
 
-    self._core = _StringCore(
-      _cocoaStringToContiguous(
-        source: cfImmutableValue, range: 0..<length, minimumCapacity: 0))
+    // Look first for null-terminated ASCII
+    // Note: the code in clownfish appears to guarantee
+    // nul-termination, but I'm waiting for an answer from Chris Kane
+    // about whether we can count on it for all time or not.
```

Comment:
Can you elaborate? Like it obviously relates to this code, I'm just not clear why it's an interesting thing. Wild guess: this is a terrible proxy for "is ASCII" if it constantly fails out on missing nul-terminators?